docs/tasks/argo.rst

```diff
@@ -0,0 +1,253 @@
+Leveraging ActionSet Creation using Argo Cron Workflows
+------------------------------------------------------------
+
+Argo Workflows enable us to schedule operations. In the Kanister project,
+Argo Cron Workflows will be used to automate the creation of ActionSets to
+backup an application at regular intervals.
+
+To summarize, ActionSets are CRDs that are used to backup or restore an application.
+The controller present in the Kanister namespace looks for ActionSets to
+execute in the given namespace.
+
+In this tutorial you will schedule the creation of a backup actionset using
+Argo Cron Workflows.
+
+
+Prerequisites
+===============
+- Kubernetes 1.16+
+- Kanister controller version 0.80.0 installed in your cluster, let's assume in
+  Namespace kanister
+- Kanctl CLI installed (https://docs.kanister.io/tooling.html#install-the-tools)
+
+Architecture
+===============
+
+.. image:: img/argo-cron-architecture.png
+
+Steps -
+===============
+
+
+Step 1 - Setup Argo
+``````````````````````
+
+Download the Argo CLI from their Releases_ page.
+
+Create a separate namespace for the Workflows.
+
+.. code-block:: bash
+
+  kubectl create ns argo
+
+For this tutorial, the minimal manifest file will be used to download
+Argo CRDs and setup their resources.
+Download the minimal manifest file from their Examples_ page.
+
+On downloading this yaml file, you may configure the managed namespaces
+as per your use case.
+You may read more about managed namespaces here - ManagedNamespaces_
+
+In this tutorial, the minimal manifest will be used as it is.
+
+.. code-block:: bash
+
+  kubectl apply -f quick-start-minimal.yaml -n argo
+
+Port-forward to view the Argo UI:
+
+.. code-block:: bash
+
+  kubectl -n argo port-forward deployment/argo-server 2746:2746
```

Comment:
Do we not have a service at this point, if there is  a service, we should port-forward to service.

Reply:
Didn't quite get this. I'm following the Argo official doc here to view their UI.

Reply:
ok, no problem. So when we try to expose an application (pod) we create a service. And then port-forward to that service. But if its from their official documentation it's ok.